docs/snippets/ov_dynamic_shapes.c

```diff
@@ -0,0 +1,254 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <openvino/c/openvino.h>
+
+void reshape_with_dynamics() {
+{
+//! [ov_dynamic_shapes:reshape_undefined]
+ov_core_t* core = NULL;
+ov_core_create(&core);
+
+ov_model_t* model = NULL;
+ov_core_read_model(core, "model.xml", NULL, &model);
+
+// Set one static dimension (= 1) and another dynamic dimension (= Dimension())
+{
+ov_partial_shape_t partial_shape;
+int64_t rank = 2;
+ov_dimension_t dims[2] = {{1, 1}, {-1, -1}};
+ov_partial_shape_create(rank, dims, &partial_shape);
+ov_model_reshape(model, "tensor_name", &partial_shape, 1); // {1,?}
```

Comment:
Real dynamic shape support should not call reshape interface, but GPU plugin doesn't enable real dynamic shape so far.

Reply:
Following with C++, I think this part need to be changed after GPU enable.